controllers/torrent/files.go

```diff
@@ -0,0 +1,78 @@
+package torrentController
+
+import (
+	"html/template"
+	"encoding/hex"
+	"net/http"
+	"strings"
+	"strconv"
+
+	"github.com/NyaaPantsu/nyaa/models/torrents"
+	"github.com/NyaaPantsu/nyaa/models"
+	"github.com/NyaaPantsu/nyaa/templates"
+	"github.com/NyaaPantsu/nyaa/utils/format"
+	"github.com/NyaaPantsu/nyaa/utils/filelist"
+	"github.com/Stephen304/goscrape"
+	"github.com/gin-gonic/gin"
+)
+
+func GetFilesHandler(c *gin.Context) {
+	id, _ := strconv.ParseInt(c.Param("id"), 10, 32)
+	torrent, err := torrents.FindByID(uint(id))	
+
+	if err != nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+	
+	
+	if len(torrent.FileList) == 0 {
+		var blankScrape models.Scrape
+		ScrapeFiles(format.InfoHashToMagnet(strings.TrimSpace(torrent.Hash), torrent.Name, GetTorrentTrackers(torrent)...), torrent, blankScrape, true)
+	}
+	
+	folder := filelist.FileListToFolder(torrent.FileList, "root")
+	templates.TorrentFileList(c, torrent.ToJSON(), folder)
+}
+
+// ScrapeFiles : Scrape torrent files
+func ScrapeFiles(magnet string, torrent *models.Torrent, currentStats models.Scrape, statsExists bool) (error, []FileJSON) {
+	if client == nil {
+		err := initClient()
+		if err != nil {
+			return err, []FileJSON{}
+		}
+	}
+	
+	t, _ := client.AddMagnet(magnet)
+	<-t.GotInfo()

```

Comment:
It might also be better to make it in a `go func()` when you try to scrap. So you don't keep the connection between the user and the server hanging. It might a while before yhe scrapper connects to a peer and gets the information.

Reply:
That itself is not a problem, since the torrent page load is not affected by this. The filelist fetching is generally done with JS, if the user doesn't have JS then a /files/:torrentid page is loaded but the user will known full well that if that page takes time to load, it's because it's loading the filelist
So it's not really necessary here